sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/client.py

```diff
@@ -762,6 +790,10 @@ def __init__(self, hostname, source, **kwargs):
         self._max_message_size = kwargs.pop("max_message_size", MAX_FRAME_SIZE_BYTES)
         self._link_properties = kwargs.pop("link_properties", None)
         self._link_credit = kwargs.pop("link_credit", 300)
+        self._timeout = kwargs.pop("timeout", 0)
+        self._timeout_reached = False
+        self._last_activity_stamp = time.time()
+        self._running_iter = False
```

Comment:
Are all these new properties _just_ used for the iterator?
Or do the other receive methods need to respect them as well?
If it's the former - then I think we should find a way to isolate these a bit...